modules/create_timepoint/php/timepoint.class.inc

```diff
@@ -109,6 +109,21 @@ class Timepoint extends \NDB_Page implements ETagCalculator
         }
         if (empty($values['psc'])) {
             unset($values['psc']);
+        } else {
+            // Suggested site is the site of the last registered visit
+            $factory = \NDB_Factory::singleton();
+            $DB      = $factory->database();
```

Comment:
You should be using `$this->loris->getDatabaseConnection()` to get a DB connection.